File: programmers/월간 코드 챌린지/시즌2/약수의 개수와 덧셈.py
def solution(left, right):
    answer = 0
    nums = dict()
    for i in range(left, right + 1):
        count = 0
        for j in range(1, i + 1):
            if i % j == 0:
                count += 1
        nums[i] = count

    for i in range(left, right + 1):
        if nums[i] % 2 == 0:
            answer += i
        else:
            answer -= i

    return answer

print(solution(13, 17)) #43
print(solution(24, 27)) #52

File: programmers/월간 코드 챌린지/시즌2/약수의개수와덧셈.js
const getDivisorCount = (num) => {
    let count = 0;
    for (let i = 1; i < num + 1; i++) {
        if (num % i === 0) {
            count += 1;
        }
    }
    return count;
}

function solution(left, right) {
    let answer = 0;
    for (let i = left; i < right + 1; i++) {
        if (getDivisorCount(i) % 2 === 0) {
            answer += i;
        } else {
            answer -= i;
        }
    }
    return answer;
}

console.log(solution(13, 17)); // 43